app/Models/OAuth2/Client.php

```diff
@@ -1749,4 +1759,28 @@ public function getOTPByValue(string $value):?OAuth2OTP{
     public function hasOTP(string $value):bool{
         return !is_null($this->getOTPByValue($value));
     }
+
+    /**
+     * @return int
+     */
+    public function getMaxAllowedUserSessions(): int
+    {
+        return $this->max_allowed_user_sessions;
+    }
+
+    public function isLimitingAllowedSessionsPerUser():bool{
+        return  $this->max_allowed_user_sessions > 0;
+    }
+
+    /**
+     * @param int $max_allowed_user_sessions
+     * @throws ValidationException
+     */
+    public function setMaxAllowedUserSessions(int $max_allowed_user_sessions): void
+    {
+        if($max_allowed_user_sessions < 0){
```

Comment:
we need to check here the client type
JS
WEB
or NATIVE
otherwise validation exception